### ISSUE TITLE
RateLimit.GetCredentialsRateLimits null reference exception

### ISSUE THREAD

**Issue by drgraduss** [linvi/tweetinvi#278]
I'm using `RateLimit.GetCredentialsRateLimits(credentials, true)` to get rate limits for the given credentials.

Sporadically I get null in return. After having a look at the library code I can see multiple possibilities when this method can return null.

        private ICredentialsRateLimits RefreshCredentialsRateLimits(ITwitterCredentials credentials)
        {
            var tokenRateLimits = GetTokenRateLimitsFromTwitter(credentials);
            _rateLimitCache.RefreshEntry(credentials, tokenRateLimits);
            return _rateLimitCache.GetCredentialsRateLimits(credentials);
        }

This method is not thread safe and between setting rate limit cache and returning it back to caller other thread might clear it. Plus `GetTokenRateLimitsFromTwitter` method can return null.

What would be best strategy to obtain credential's rate limits?

**Comment by linvi** [comment]
I will investigate and get back to you.

**Comment by linvi** [comment]
So I just verified and this cannot be a thread issue because the `RateLimitManager` has a per thread injection strategy. More information to come...

**Comment by linvi** [comment]
So I reviewed the code, made some changes but I could not find anything wrong about it. The only thing that I can think of is that you tried to get the RateLimits too often and it ended up returning null.

Would you please share your code?

Also please note that 0.9.13.0 has improved the accuracy of the cached information which are now updated every time a RateLimited with the value returned in the headers.

**Comment by linvi** [comment]
Please provide some feedback before the end of the day as I would like to release 0.9.14.0 tonight.
Otherwise I will have to postpone this to release 1.0.

**Comment by drgraduss** [comment]
I have nothing special in terms of querying rate limits apart from I'm doing it in parallel and rate limits for the same credentials might be requested concurrently. Thanks for looking into this.

**Comment by linvi** [comment]
Ok. The code changes in 0.9.14.0 might help you but I cannot guarantee that without having proper testing over your code.

I have on my side worked out with multithreaded application accessing the RateLimit for the same credentials without any problem.

I am therefore closing this issue for now.

Please let me know if you still encounter the same problem.

**Comment by drgraduss** [comment]
Okay. I will update and keep an eye on this issue. Thanks again.
